src/Registry/Utils/PostObject.php

```diff
@@ -198,14 +198,15 @@ protected static function get_connections( WP_Post_Type $post_type_object ) {
 					),
 					'resolve'        => static function ( Post $post, $args, AppContext $context, ResolveInfo $info ) {
 						$taxonomies = \WPGraphQL::get_allowed_taxonomies();
-						$terms      = wp_get_post_terms( $post->ID, $taxonomies, [ 'fields' => 'ids' ] );
+						$object_id = true === $post->isPreview && ! empty( $post->parentDatabaseId ) ? $post->parentDatabaseId : $post->ID;
 
-						if ( empty( $terms ) || is_wp_error( $terms ) ) {
+						if ( empty( $object_id ) || ! absint( $object_id ) ) {
 							return null;
 						}
-						$resolver = new TermObjectConnectionResolver( $post, $args, $context, $info, $taxonomies );
-						$resolver->set_query_arg( 'include', $terms );
 
+						$resolver = new TermObjectConnectionResolver( $post, $args, $context, $info, $taxonomies );
+						$resolver->set_query_arg( 'object_ids', absint( $object_id ) );
+						$resolver->set_query_arg( 'taxonomies', $taxonomies );
```

Comment:
1. If we're already updating this callback, better to insert these via `$args['where]` than to overload it after the fact with `->set_query_arg()`
2. Not by my desk to confirm, but `set_query_arg( 'taxonomies' )` seems redundant when it's also passed to the constructor.

Reply:
ah yes! I think I added the set_query_arg first, changed my mind, but didn't remove it 🤦🏻 

I made this PR from the Github UI mostly so I had a placeholder. Will update the PR locally now.